Extension/src/LanguageServer/configurations.ts

```diff
@@ -1343,9 +1356,29 @@ export class CppProperties {
             }
             curTextStartOffset = configStart + 1;
             curText = curText.substr(curTextStartOffset); // Remove earlier configs.
-            const nameEnd: number = curText.indexOf(":");
+            let nameEnd: number = curText.indexOf(":");
             curTextStartOffset += nameEnd + 1;
             curText = curText.substr(nameEnd + 1);
+            // Check if the config name is unique, before removing later configs
+            const nameRegex: RegExp = new RegExp(`{\\s*"name"\\s*:\\s*"${escapeStringRegExp(currentConfiguration.name)}"`);
```

Comment:
I had commented without starting a 'review', so I can't check if my topic was considered 'resolved' or not.  Opening this to track, just in case.  :)

This is where the `currentConfiguration` was used to only detect duplicates of only the current config's name.  The local `configurations` var contains the text as parsed JSON object(s), and could be used to detect all duplicate names.  Something similar to the code below would still be needed to determine the exact positions to put squiggles.

Reply:
![image](https://user-images.githubusercontent.com/16143955/104517887-ade27480-55ab-11eb-98ef-6f46183df2d8.png)
this is the latest update

Reply:
Shouldn't we also squiggle the first duplicate?  If the other duplicates happen to be scrolled out of view when the user changes the name of an earlier config, they may not notice.

Reply:
For example, when there are duplicate field names in a JSON object, in a .json file, VS Code will squiggle all duplicates names.